packages/core/utils/src/dml/helpers/entity-builder/query-builder.ts

```diff
@@ -0,0 +1,57 @@
+import { QueryCondition, QueryValue, SimpleQueryValue } from "@medusajs/types"
+
+/*
+  When creating indexes on the entity, we provide a basic query builder to generate
+  the SQL for where query upon the index. Since this is not a full query builder, 
+  the onus is upon the user to ensure that the SQL is accurately generated.
+
+  Examples:
+
+  { where: { column: null } }
+  { where: { column: { $ne: null } } }
+  { where: { boolean_column: true } }
+  { where: { column: "value", another_column: { $ne: 30 }, boolean_column: true } }
+*/
+export function buildWhereQuery(queryObj: QueryCondition): string {
+  const conditions: string[] = []
+
+  for (const [key, value] of Object.entries(queryObj)) {
+    if (isQueryCondition(value)) {
+      conditions.push(buildWhereQuery(value))
+    } else {
+      conditions.push(buildCondition(key, value as QueryValue))
+    }
+  }
+
+  return conditions.join(" AND ")
+}
+
+function isQueryCondition(value: any): value is QueryCondition {
+  return typeof value === "object" && value !== null && !("$ne" in value)
+}
+
+function buildCondition(key: string, value: QueryValue): string {
```

Comment:
Suggestion: i would add the boolean support through is true is false because for nullable boolean you can have null and != false would gives you true and null boolean.